tests/supervisors/test_signal.py

```diff
@@ -0,0 +1,102 @@
+import asyncio
+import signal
+from asyncio import Event, Task
+from functools import partial
+
+import httpx
+import pytest
+
+from tests.utils import run_server
+from uvicorn import Server
+from uvicorn.config import Config
+
+
+def set_event(event: Event, task: Task):
+    event.set()
+
+
+@pytest.mark.anyio
+async def test_sigint_finish_req(unused_tcp_port: int):
+    """
+    Test that a request that is sent, sigint is sent, but request is finished and not cancelled
+    """
+
+    server_event = Event()
+
+    async def wait_app(scope, receive, send):
+        await send({"type": "http.response.start", "status": 200, "headers": []})
+        await send({"type": "http.response.body", "body": b"start", "more_body": True})
+        await server_event.wait()
+        await send({"type": "http.response.body", "body": b"end", "more_body": False})
```

Comment:
Btw @adriangb, thanks for the talk @ PyCon - I learned about the `more_body` key 😊